### PR TITLE
ENH: Porting CCD & CCDD tests to TestCases

### PR DESCRIPTION
**Pull request by thomas-rocke**

### CODE REVIEW

File: tests/test_dislocation.py [hunk 1]
```diff
@@ -107,7 +106,7 @@ def test_elastic_constants_EAM(self):
         pot_path = os.path.join(test_dir, pot_name)
         calc_EAM = EAM(pot_path)
         obtained_values = sd.get_elastic_constants(calculator=calc_EAM,
-                                                   delta=1.0e-3)
+                                                   delta=1.0e-3, verbose=False)
```
Comment (jameskermode) [inline tests/test_dislocation.py:141]: not necessary to reduce verbosity when using pytest

File: tests/test_dislocation.py [hunk 2]
```diff
@@ -904,5 +589,430 @@ def test_mixed_dislocation_adsl(self):
...
+                    np.testing.assert_array_almost_equal(method_disps, base_displacements)
+                except AssertionError as e:
+
+                    raise AssertionError(f"Displacements from {method} did not match {base_method}")
+
+class TestCubicCrystalDissociatedDislocation(TestCubicCrystalDislocation):
+    def __init__(self, disloc_cls):
+        TestCubicCrystalDislocation.__init__(self, disloc_cls)
+
+    def set_test_name(self):
```
Comment (jameskermode) [inline tests/test_dislocation.py]: make clear this is part of __init__

File: tests/test_dislocation.py [hunk 3]
```diff
@@ -904,5 +589,430 @@ def test_mixed_dislocation_adsl(self):
...
+        bulk, quad = d.build_quadrupole(glide_separation=14, partial_distance=4, verbose=False)
+
+
+
+
+##############
+# Build correct unittest test suite
+##############
+
+def load_tests(loader, tests, pattern):
```
Comment (jameskermode) [inline tests/test_dislocation.py]: check if this gets called by `pytest .` from matscipy root

**Review by jameskermode — APPROVED** [review]

File: tests/test_dislocation.py (hunk 3 above)
Comment (thomas-rocke) [reply tests/test_dislocation.py]: Added a pytest-based variant to `MatSciPyTestCase` (called `MatSciPyTestFixure`), which does not inherit from `unittest.TestCase` - this means that the class is directly `parametrize`-able. Not all features of `TestCase` implemented (as many weren't needed for this set of tests), but there is coverage of the custom `assertAtomsEqual` function etc.

**Pull request merged**